lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -1262,6 +1242,8 @@ def __encode_invocation(self, invocation, **kwd):
 class FastAPIWorkflows:
     service: WorkflowsService = depends(WorkflowsService)
     invocations_service: InvocationsService = depends(InvocationsService)
+    workflows_manager: WorkflowsManager = depends(WorkflowsManager)
```

Comment:
If you want to do some more refactoring I'd split this into a workflows class and an invocations class, I think they're logically distinct.

Reply:
You already have access to the `WorkflowsManager` in the `WorkflowsService` above (`self.service.workflows_manager`) :)
The service encapsulates every related manager that your API needs. One of the main functions of the service layer is to encapsulate all the API logic so the endpoint definition is as thin as possible and deals mostly with parameter conversion and documentation annotations.

Reply:
Yeah, sorry, what I meant is this https://github.com/galaxyproject/galaxy/pull/14794/files#r999056504